client/src/utils/waitOnUrl.ts

```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+
+export function waitOnUrl(url: string, timeout: number, interval: number, log = console.info): Promise<unknown> {
+  let timeoutTimer: NodeJS.Timeout;
+  let timedOut = false;
+  let timeoutResolve: (...args: any) => any;
+  function _waitOnUrl(url: string, whenAvailable: (...args: any) => void) {
+    setTimeout(
+      () => {
+        void axios.get(url, {
+          timeout: interval
+        }).then((value) => {
+          clearTimeout(timeoutTimer);
+          timeoutResolve();
+          whenAvailable();
+        }).catch((error) => {
+          if (timedOut) return;
+          log('.');
+          _waitOnUrl(url, whenAvailable);
```

Comment:
we should only keep waiting if http code was 503 (I think), any other kind of problem is likely unrecoverable, so no need to wait any further. Error should bubble up to user.